### PR TITLE
Changing how artifact visibility works

### PR DESCRIPTION
**Pull request by antgonza**
I'm almost sure some of the tests will fail and I can fix later but want to see if you guys agree with this approach before expending time fixing them. 

@wasade and @josenavas , what do you think?

### CODE REVIEW

File: qiita_db/artifact.py [hunk 1]
```diff
@@ -617,19 +619,20 @@ def visibility(self, value):
         only applies when the new visibility is more open than before.
         """
         with qdb.sql_connection.TRN:
+            # In order to correctly propagate the visibility we need to find
+            # the root of this artifact and then propagate to all the artifacts
+            root = self.artifact
```
Comment (josenavas) [inline qiita_db/artifact.py]: What is self.artifact? 

Anyways, you can simplify this by:

```python
sql = "SELECT * FROM qiita.find_artifact_roots(%s)"
TRN.add(sql, [self.id])
root = TRN.execute_fetchlast()
```

File: qiita_db/support_files/patches/python_patches/53.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+for rd in raw_data:
+    # getting the most open visibility of all the children in the pipeline
+    children = rd.descendants.nodes()
+    vis = [a.visibility for a in children]
+    vis.append(rd.visibility)
+
+    new_vis = 'sandbox'
+    if 'public' in vis:
+        new_vis = 'public'
+    elif 'private' in vis:
```
Comment (wasade) [inline qiita_db/support_files/patches/python_patches/53.py]: are there situations where an artifact was bumped to private but where the user expects all other artifacts to remain sandboxed?

File: qiita_db/artifact.py (hunk 1 above)
Comment (antgonza) [inline qiita_db/artifact.py]: brain fart, thanks! Will fix ...

File: qiita_db/support_files/patches/python_patches/53.py (hunk 2 above)
Comment (antgonza) [inline qiita_db/support_files/patches/python_patches/53.py]: in praxis no ... at least based on what we have seen so far ...

File: qiita_db/artifact.py [hunk 3]
```diff
@@ -260,9 +260,11 @@ def create(cls, filepaths, artifact_type, name=None, prep_template=None,
 
         Notes
         -----
-        The visibility of the artifact is set by default to `sandbox`
-        The timestamp of the artifact is set by default to `datetime.now()`
-        The value of `submitted_to_vamps` is set by default to `False`
+        The visibility of the artifact is set by default to `sandbox` if
+        prep_template is passed but if parents is passed we will inherit the
+        most closed visibility.
```
Comment (ElDeveloper) [inline qiita_db/artifact.py]: most closed -> closest?

File: qiita_db/artifact.py [hunk 4]
```diff
@@ -260,9 +260,11 @@ def create(cls, filepaths, artifact_type, name=None, prep_template=None,
 
         Notes
         -----
-        The visibility of the artifact is set by default to `sandbox`
-        The timestamp of the artifact is set by default to `datetime.now()`
-        The value of `submitted_to_vamps` is set by default to `False`
+        The visibility of the artifact is set by default to `sandbox` if
+        prep_template is passed but if parents is passed we will inherit the
```
Comment (ElDeveloper) [inline qiita_db/artifact.py]: `is passed but` -> `is passed and if`?

**Comment by antgonza** [comment]
This is ready for actual review ...

**Comment by antgonza** [comment]
@wasade, @ElDeveloper, @josenavas, wanna take a look?

File: qiita_db/artifact.py (hunk 3 above)
Comment (ElDeveloper) [inline qiita_db/artifact.py]: I understand now, what you mean here.

File: qiita_db/artifact.py (hunk 4 above)
Comment (ElDeveloper) [inline qiita_db/artifact.py]: I think I also understand this as well.

**Pull request merged**